firehose/firehose.go

```diff
@@ -0,0 +1,242 @@
+		c.region = defaultRegion
+	}
+
+	if os.Getenv("LOCALSTACK_HOSTNAME") != "" {
+		c.endPoint = os.Getenv("LOCALSTACK_HOSTNAME")
+	}
+
+	customResolver := aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+		if c.endPoint != "" {
+			return aws.Endpoint{
+				URL:           fmt.Sprintf("http://%s:4566", c.endPoint),
+				SigningRegion: c.region,
+			}, nil
+		}
+
+		// returning EndpointNotFoundError will allow the service to fallback to it's default resolution
+		return aws.Endpoint{}, &aws.EndpointNotFoundError{}
+	})
+
+	cfg, err := config.LoadDefaultConfig(context.Background(),
+		config.WithRegion(c.region),
+		config.WithEndpointResolver(customResolver),
+	)
+	if err != nil {
+		return nil, err
+	}
+	c.firehose = firehose.NewFromConfig(cfg)
+
+	return c, nil
+}
+
+// CreateDeliveryStream creating firehose delivery stream channel
+// You must provide channel name as required parameter
+// If channel created successfully it will return nil else it will return error
+func (c *ClientProvider) CreateDeliveryStream(channel string) error {
+	params := &firehose.CreateDeliveryStreamInput{
+		DeliveryStreamName: aws.String(channel),
+		DeliveryStreamType: types.DeliveryStreamTypeDirectPut,
+	}
+	_, err := c.firehose.CreateDeliveryStream(context.Background(), params)
+	return err
+}
+
+// PutRecordBatch is operation for Amazon Kinesis Firehose
+// Writes multiple data records into a delivery stream in a single call, which
+// can achieve higher throughput per producer than when writing single records.
+//
+// Each PutRecordBatch request supports up to 500 records. Each record in the
+// request can be as large as 1,000 KB (before 64-bit encoding), up to a limit
+// of 4 MB for the entire request.
+//
+// You must specify the name of the delivery stream and the data record when
+// using PutRecord. The data record consists of a data blob that can be up to
+// 1,000 KB in size.
+//
+// The PutRecordBatch response includes a map of failed records.
+// Even if the PutRecordBatch call succeeds
+//
+// Data records sent to Kinesis Data Firehose are stored for 24 hours from the
+// time they are added to a delivery stream as it attempts to send the records
+// to the destination. If the destination is unreachable for more than 24 hours,
+// the data is no longer available.
+//
+// Don't concatenate two or more base64 strings to form the data fields of your
+// records. Instead, concatenate the raw data, then perform base64 encoding.
+func (c *ClientProvider) PutRecordBatch(channel string, records []interface{}) ([]*PutResponse, error) {
+	ch := make(chan *chanPutResponse)
+	chunk := make([]interface{}, 0)
+	requestCounter := 0
+	for record := range records {
+		r, err := json.Marshal(record)
+		if err != nil {
+			return []*PutResponse{}, err
+		}
+		if len(r) > 1020000 {
+			return []*PutResponse{}, errors.New("record exceeded the limit of 1 mb")
```

Comment:
we should also handle slicing the chunks into the desired sizes instead of throwing this error